### DISCUSSION TITLE
SED child stuck with the parent hours after parent clears the child context

### DISCUSSION THREAD

**Discussion by sarveshkumarv3** [openthread/openthread#9182] [Q&A] [answered]
**Describe the bug** A clear and concise description of what the bug is.

1. Parent removes the child (SED) from its child table 
2. Parent still sends the ACK for the DataRequest from the child 
3. Child uses the ACK as keep-alive (enhanced keep alive) and does not transmit a Child Update Request 
4. Child still think parent link is good and does not trigger new parent request

**Expected behavior** A clear and concise description of what you expected to happen.

Child should detect that parent has removed the child from its table and attempt to attach 

**Console/log output** If applicable, add console/log output to help explain your problem.

Only parent side logs are available. No child update request seen from SED for around 5-6 hours

19:44:20.416 [I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:22, seqnum:220, type:Cmd(DataReq), src:0x8001, dst:0x8000, sec:yes, ackreq:yes, fp:no, fc:555884, radio:15.4
19:44:22.950 [I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:22, seqnum:221, type:Cmd(DataReq), src:0x8001, dst:0x8000, sec:yes, ackreq:yes, fp:no, fc:555885, radio:15.4
19:44:25.506 [I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:22, seqnum:222, type:Cmd(DataReq), src:0x8001, dst:0x8000, sec:yes, ackreq:yes, fp:no, fc:555886, radio:15.4
19:44:28.011 [I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:22, seqnum:223, type:Cmd(DataReq), src:0x8001, dst:0x8000, sec:yes, ackreq:yes, fp:no, fc:555887, radio:15.4
19:44:30.533 [I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:22, seqnum:224, type:Cmd(DataReq), src:0x8001, dst:0x8000, sec:yes, ackreq:yes, fp:no, fc:555888, radio:15.4
19:44:33.075 [I] Mac-----------: Frame rx failed, error:UnknownNeighbor, len:22, seqnum:225, type:Cmd(DataReq), src:0x8001, dst:0x8000, sec:yes, ackreq:yes, fp:no, fc:555889, radio:15.4

**Additional context** Add any other context about the problem here.

In this testing the child (third-party accessory) may have child supervision disabled. 

Enabling child supervision might help to recover if no message from parent is received which is in turn discussed in ([SPEC-1046](https://threadgroup.atlassian.net/browse/SPEC-1046)) 

However I think there is a mismatch in terms of spec definition still (for a legacy device which may not update firmware to enable child supervision but which has enhanced keep alive enabled and uses data poll for keep alive). 

I am interpreting the below from specification  section 4.6.3 in section 1.3.0  

**Rx-off-when-idle Children SHOULD also periodically resynchronize by sending a Child Update Request to their parent to avoid cases where the MAC Data Request is acknowledged erroneously by the receiver.**

However from OT code, it seems like we do not send Child Update Request for SED

Mle::HandleMessageTransmissionTimer
    //  - Sending periodic keep-alive "Child Update Request" messages on a non-sleepy (rx-on) child.

        // Keep-alive "Child Update Request" only on a non-sleepy child
        VerifyOrExit(IsChild() && IsRxOnWhenIdle());

Can you please help to clarify if the above scenario is a possibility with 1.3.0 and if so can we solve from parent side (assuming child cannot support child supervision)?

**Comment by abtink** [comment]
> However I think there is a mismatch in terms of spec definition still (for a legacy device which may not update firmware to enable child supervision but which has enhanced keep alive enabled and uses data poll for keep alive).
> 
> I am interpreting the below from specification section 4.6.3 in section 1.3.0
> 
> Rx-off-when-idle Children SHOULD also periodically resynchronize by sending a Child Update Request to their parent to avoid cases where the MAC Data Request is acknowledged erroneously by the receiver.

Seems like the SED was built without OT "child supervision" `OT_CHILD_SUPERVISION` feature and this is the root cause of the issue.

The OT "child supervision" was the feature that ensures that SED child would send "Child Update Request" periodically (if not supervised by its parent for whatever reason).  

Since https://github.com/openthread/openthread/pull/7993 (and SPEC-1046) it is required by spec (so `OT_CHILD_SUPERVISION` flag was removed).

**Comment by jwhui** [comment, marked as answer]
> can we solve from parent side (assuming child cannot support child supervision)?

I believe the only possible mitigation is for the parent to stop sending ACKs and the only way to do that is by changing the parent's RLOC16 or disabling its radio. Both are pretty disruptive and it's not clear how to do that while protecting against DoS attacks.

**Comment by sarveshkumarv3** [comment]
@jwhui @abtink raised JIRA [SPEC-1205] to request your thoughts about not responding to data request if child entry not present (as I  felt it is more of a spec discussion). Thanks.
